### ISSUE TITLE
podman run --user=NONEXISTENT-UID : fails to start

### ISSUE THREAD

**Issue by edsantiago** [projectatomic/libpod#631]
podman seems to be performing UID validation when run with `--user=UID` and UID is not present in the image's `/etc/passwd`:
```
# podman run --rm --user=123456 fedora id
user: unknown user error looking up user with UID 123456
```
docker happily runs the container:
```
# docker run --rm --user=123456 fedora id
uid=123456 gid=0(root) groups=0(root)
```
I lean toward thinking that podman's behavior is correct. Filing this for visibility and for greater discussion.

**Comment by rhatdan** [comment]
We should allow it to run as long as it is a valid UID IE A number.  Then we run as a number.  If it is not a number we should look it up in the /etc/passwd. Seems like we are always looking up the Username in /etc/passwd.

There was a bug in Docker where a container image could create a user called "1234" and map it to UID=0, we should not read /etc/passwd if the user is a UID versus a UserName.

@umohnani8 Could you fix this.

**Comment by umohnani8** [comment]
Yup!

**Comment by mheon** [comment]
Closed by #652

**Comment by umohnani8** [comment]
Fixed in #652.
